src/MessageList/AttachmentBar.vala

```diff
@@ -69,8 +69,8 @@ public class Mail.AttachmentBar : Gtk.FlowBox {
     }
 
     private void show_attachment (Camel.MimePart attachment_part) {
-        var dialog = new Mail.AttachmentDialog (attachment_part);
-        dialog.transient_for = get_toplevel () as Gtk.Window;
+        var dialog = new Mail.OpenAttachmentDialog ((Gtk.Window) get_toplevel (), attachment_part);
```

Comment:
Here the `as` keyword is important, if it's not a Gtk.Window we don't want to use it.